js/src/views/Signer/store.js

```diff
@@ -49,8 +49,13 @@ export default class SignerStore {
   }
 
   @action unsubscribe () {
-    if (this._timeoutId) {
-      clearTimeout(this._timeoutId);
+    const self = this;
+
+    if (self.subscribed) {
+      if (!self.subscriptionId) {
```

Comment:
You can simplify the code by saving the promise itself as subscription and then resolving it in unsubscribe.
This works fine when the id does not need to get exposed.
[E.g. like here in the bonds library](https://github.com/paritytech/oo7-parity/blob/kaikun/src/index.js#L105)